pkg/containers/containers.go

```diff
@@ -90,6 +90,47 @@ func (c *Containers) procMountsCgroups() error {
 	return nil
 }
 
+//returns the ContainerID of a given mountpoint cgourp
+func getContainerIdFromTaskContainer(taskContainer *Containers, cgroup string) string {
+	taskContainer.procMountsCgroups()
+	path := fmt.Sprintf("%s%s", taskContainer.cgroupMP, cgroup)
+	containerId, _ := getContainerIdFromCgroup(path)
+	return containerId
+}
+
+// gets a containerID from a given task or process directory path
+func GetContainerIdFromTaskDir(taskPath string) (string, error) {
+	containerId := ""
+	taskPath = fmt.Sprintf("%s/cgroup", taskPath)
+	cgroupFile, err := os.Open(taskPath)
+	if err != nil {
+		return containerId, err
+	}
+	defer cgroupFile.Close()
+	scanner := bufio.NewScanner(cgroupFile)
+	taskContainer := InitContainers()
+	for scanner.Scan() {
+		line := strings.Split(scanner.Text(), ":")
+		subsys := line[1]
+		cgroup := line[2]
+		if taskContainer.cgroupV1 {
+			if cgroup != "/" && subsys != "cpuset" {
+				containerId = getContainerIdFromTaskContainer(taskContainer, cgroup)
+				break
+			}
+		} else {
+			line := scanner.Text()
+			cgroupV2Suref := "0::/"
+			if strings.HasSuffix(line, cgroupV2Suref) {
+				continue
+			}
+			containerId = getContainerIdFromTaskContainer(taskContainer, cgroup)
+			break
+		}
+	}
```

Comment:
You can just loop over the lines, sending each line to getContainerIdFromCgroup(), and break if container id is not empty. This should work for both cgroup v1 and cgroup v2, without even having to build the path string like you do in getContainerIdFromTaskContainer